app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/LocalizationActivity.java

```diff
@@ -0,0 +1,142 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.plugins.localization.LocalizationPlugin;
+import com.mapbox.mapboxsdk.plugins.testapp.R;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class LocalizationActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+  @BindView(R.id.mapView)
+  MapView mapView;
+
+  private LocalizationPlugin localizationPlugin;
+  private boolean mapIsLocalized;
+  private String ENGLISH_LANGUAGE_CODE = "en";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_localization);
+    ButterKnife.bind(this);
+    mapIsLocalized = true;
+    Toast.makeText(this, R.string.change_language_instruction, Toast.LENGTH_LONG).show();
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    localizationPlugin = new LocalizationPlugin(mapboxMap);
+  }
+
+  @OnClick(R.id.localize_fab)
+  public void localizeToggleFab() {
+    if (mapIsLocalized) {
+      localizationPlugin.setMapTextLanguage(ENGLISH_LANGUAGE_CODE);
+      Toast.makeText(this, R.string.map_not_localized, Toast.LENGTH_SHORT).show();
+      mapIsLocalized = false;
+    } else {
+      localizationPlugin.setMapTextLanguage(localizationPlugin.getDeviceLanguage());
```

Comment:
If the device language happens to be a language that the Mapbox Streets source doesn’t support, such as Thai, this plugin will end up changing all the layers to `{name_th}`, producing undesired results. This plugin needs to check that the device language is [supported](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview), falling back to either English or the local language. (The iOS and macOS map SDKs fall back to English, whereas the Mapbox GL Language plugin for GL JS falls back to a default language that the developer specifies as an option.)

Reply:
I believe that https://github.com/mapbox/mapbox-plugins-android/pull/74/commits/af59d951965aa060e8d0e91726ac216d92966759  now fixes this. I've added a preferred backup language parameter for the initialization of the plugin.

https://github.com/mapbox/mapbox-plugins-android/blob/af59d951965aa060e8d0e91726ac216d92966759/plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java#L39

I also added a check to make sure that a language is supported by OSM
https://github.com/mapbox/mapbox-plugins-android/blob/af59d951965aa060e8d0e91726ac216d92966759/plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java#L87-L97

If the check detects that the device language isn't supported, then the backup language is checked and set. English is the final fall back if the *backup* language isn't supported by OSM either

https://github.com/mapbox/mapbox-plugins-android/blob/af59d951965aa060e8d0e91726ac216d92966759/plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java#L130-L138